### PR TITLE
PUT || Edit User

### PR DESCRIPTION
**Pull request by ElhamFadel**

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const users = require('./users');
 
 router.use('/users', users);
 router.use('/estate', estate);
+router.use('/users', users);
```
Comment (aaamra) [inline server/routes/index.js:8]: why you added this line ?

its already exists in line 6

File: server/controllers/users/putAgent.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const { putAgent } = require('../../database/quieres');
+const editAgentSchema = require('../../utils/validation/editAgentSchema');
+
+module.exports = async (req, res, next) => {
+  try {
+    const { userId } = req.params;
+    const { error, value } = await editAgentSchema.validate({ ...req.body, userId });
```
Comment (aaamra) [inline server/controllers/users/putAgent.js:7]: use `validateAsync` instead of `validate`

File: server/test/index.test.js [hunk 3]
```diff
@@ -250,3 +251,49 @@ describe('test signup endpoint with all cases ', () => {
     });
   });
 });
+
+describe('test Edit Agent data /users/:iduser  ', () => {
```
Comment (aaamra) [inline server/test/index.test.js:255]: Great work at tests

but i suggest you to create multiple files for each group of related tests
its easier to read 

this file is about 300 lines

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by emanNahed — APPROVED** [review]

File: server/routes/index.js [hunk 4]
```diff
@@ -1,12 +1,13 @@
 const router = require('express').Router();
-const auth = require('./auth');
 const estate = require('./estate');
 const users = require('./users');
 const admins = require('./admin');
+const { logout, login } = require('../controllers');
 
-router.use('/users', users);
+router.get('/logout', logout);
+router.post('/login', login);
```
Comment (Mu7ammadAbed) [inline server/routes/index.js:8]: This belongs to the user route

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by emanNahed — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Pull request merged**